spec/models/bus_stops_route_spec.rb

```diff
@@ -35,4 +35,57 @@
       end
     end
   end
+  describe 'import' do
+    let(:main_variant) { %w[A B C D E] }
+    before :each do
+        @stop_hash = { route_dir: { key_1: main_variant, key_2: other_variant } }
+    end
+    let(:resequence!) { BusStopsRoute.import(@stop_hash) }
+    let(:other_variant) { %w[A B C D E] }
+    context 'all stops are in longest variant' do
+      it 'preserves sequence of longest variant' do
+        resequence!
+        expect(@stop_hash[:route_dir]).to eql %w[A B C D E]
+      end
+    end
+    context 'stop (but not first stop) is not in longest variant' do
+      context 'one stop is in other variant' do
+        let(:other_variant) { %w[A F E] }
```

Comment:
Isn't this two stops in the other variant?

Reply:
@jforgue is confused by my test descriptions